### PR TITLE
Proper indentation for template literals

### PR DESCRIPTION
**Pull request by vjeux**
Right now, expressions inside of template literals are have the level of indentation of the start of the template literal. But oftentimes, expressions are nested inside of template literals and therefore the expression should have this level of nesting.

The heuristic i'm using to decide when to use the template literal nesting is if it's not the first line and if there isn't another expression before on the same line.

Original:
```js
module.exports = Relay.createContainer(
  // ...
  {
    fragments: {
      nodes: ({solution_type, time_frame}) => Relay.QL`
        fragment on RelatedNode @relay(plural: true) {
          __typename
          ${OptimalSolutionsSection
            .getFragment(
              'node',
              {solution_type, time_frame},
            )
          }
        }
      `,
    },
  },
);
```

Before:
```js
module.exports = Relay.createContainer(
  // ...
  {
    fragments: {
      nodes: ({ solution_type, time_frame }) => Relay.QL`
        fragment on RelatedNode @relay(plural: true) {
          __typename
          ${OptimalSolutionsSection.getFragment("node", {
        solution_type,
        time_frame
      })}
        }
      `
    }
  }
)
```

After:
```js
module.exports = Relay.createContainer(
  // ...
  {
    fragments: {
      nodes: ({ solution_type, time_frame }) => Relay.QL`
        fragment on RelatedNode @relay(plural: true) {
          __typename
          ${OptimalSolutionsSection.getFragment("node", {
            solution_type,
            time_frame
          })}
        }
      `
    }
  }
);
```

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -1475,9 +1475,43 @@ function genericPrintNoParens(path, options, print, args) {
...
+          const index = value.lastIndexOf('\n');
+          const tabWidth = options.tabWidth;
+          if (index !== -1) {
+            for (let i = index + 1; i < value.length; ++i) {
+              if (value[i] === '\t') {
+                // Tabs behave in a way that they are aligned to the nearest
+                // multiple of tabWidth:
+                // 0 -> 4, 1 -> 4, 2 -> 4, 3 -> 4
+                // 4 -> 8, 5 -> 8, 6 -> 8, 7 -> 8 ...
+                size = Math.floor(((size + tabWidth) / tabWidth) * tabWidth);
```
Comment (lydell) [inline src/printer.js]: This line is equivalent to `size += tabWidth`, isn't it? I think you might be looking for `size = size + tabWidth - size % tabWidth`.

Comment (vjeux) [inline src/printer.js]: No it's not, I floor it in between. So if `size === 3`, it's going to be `4` and not `7`.

Your computation does the same thing and is smaller, i'll use it instead :) Thanks!

Comment (vjeux) [inline src/printer.js]: Hmmm, actually it is... I must have screwed up when copy and pasting from the console...

**Comment by jlongster** [comment]
Thanks for taking this on! This is a hard one, but I think this is the right approach. I ran out of time today but I'll take a closer look first thing tomorrow!

**Comment by vjeux** [comment]
I didn't test the tabs use case and it showed. Now it's all working fine and I added a test for tabs.

**Pull request merged**